src/lib/reasoners/satml_frontend.ml

```diff
@@ -1350,6 +1357,8 @@ module Make (Th : Theory.S) : Sat_solver_sig.S = struct
            (* don't attempt to compute a model if timeout before
               calling unsat function *)
            i_dont_know env (Timeout Assume)
+         | Util.Step_limit_reached _ ->
+           env
```

Comment:
I believe that this is still leaking a potentially invalid environment into the wild without protecting it with an unknown reason. This should raise i_dont_know and it should be caught in handle_sat_exn in the frontend to set the state there to  `Unknown and following operations will be protected by wrap_f.

Reply:
If we raise `I_dont_know` we will answer `unknown` during an assertion; is that really what we want to do? I would rather update the environment with the correct `unknown_reason`

Reply:
> If we raise I_dont_know we will answer unknown during an assertion

If we raise `I_dont_know` I think it should be caught by the frontend which will update the unknown reason as needed, and only display `unknown` once the solving loop is done processing, but worth checking.

Reply:
The unknown reason is on the sat solver side

Reply:
Ah right — yeah returning the env with `Step_limit` unknown reason is what needs to happen then.